### PR TITLE
Adjust tests related to /etc/shadow

### PR DESCRIPTION
**Pull request by smcv**
* test-run: Skip a test if we can read /etc/shadow
    
    Ordinarily, we would not be able to read /etc/shadow if we're not uid 0;
    but when building in a sysroot owned by the current user (for example
    by setting it up using bwrap, as steam-runtime-tools does), we might
    actually be able to read it. Skip the assertion that we cannot read it
    in this case.

* test-run: Add another assertion that we cannot read /etc/shadow
    
    The goal of this assertion was to demonstrate that a setuid bwrap does
    not give us access to otherwise unreadable files, but if we want to
    check that, we should probably be looking at the bind-mount destination
    instead of the source file.
    
    Leave the old assertion in too, just in case *that* fails.

---

@RyuzakiKK will probably find this useful.

### CODE REVIEW

**Review by RyuzakiKK — APPROVED** [review]
I reviewed this patch as part of a steam-runtime-tools MR.
Not a maintainer but LGTM.

**Comment by smcv** [comment]
Timed out waiting for review from another maintainer, merging based on positive review from my colleague @RyuzakiKK.

**Pull request merged**
